### ISSUE TITLE
[JAWS + IE11] "escape" key causes virtual cursor to get stuck after closing a dialog

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/john-travoltage#247]
Seen on Windows 10, IE11 + JAWS. For phetsims/tasks#831. Found by the illustrious @phet-steele.

When the escape key is used to close a dialog in JAWS, the virtual cursor seems to get stuck in the sim title (or something).  After closing a dialog with escape key, when moving the virtual cursor the only thing that can be read is "John Travoltage".  In this state, nothing can be interacted with. When you try to press a button, focus leaves the sim and JAWS reads "John Travoltage" instead.

**Comment by jessegreenberg** [comment]
This bug makes the sim unusable, so we should definitely find a workaround.

**Comment by jessegreenberg** [comment]
My guess is that JAWS is not correctly updating the accessibility tree when the dialog is un-hidden. Sometimes this will happen if you click on the button with "enter" or "spacebar". And sometimes this doens't happen when you close the dialog with "Escape".

**Comment by jessegreenberg** [comment]
I am wondering if this is related to https://github.com/phetsims/a11y-research/issues/30 - When a parent is un-hidden, children aren't added back into the accessibility tree. Is that similar to what is going on here?

**Comment by jessegreenberg** [comment]
The problem goes away when the phet menu is opened, is that forcing the accessibility tree to update?

**Comment by jessegreenberg** [comment]
Removing the entire subtree of accessible content and adding it back in depending on whether or not the dialog is open seems to fix this issue.

**Comment by jessegreenberg** [comment]
That might be the only way to guarantee that this works in JAWS, but it will be much more performance intensive if we have to hide everything this way.

**Comment by jobara** [comment]
@jessegreenberg have you tried refreshing JAWS virtual buffer. You should be able to do that as a user with the JAWS key + Escape. This might help to give clues to the issue.

**Comment by jessegreenberg** [comment]
Thanks @jobara, I didn't know you could do that! @phet-steele and I just tried to do that and the problem is still there. When you refresh the virtual buffer does it update everything in the accessibility tree?

**Comment by jessegreenberg** [comment]
The above commit fixes the problem, but we should verify performance on a tablet.

**Comment by jessegreenberg** [comment]
I just tested performance on http://www.colorado.edu/physics/phet/dev/html/john-travoltage/1.3.0-dev.37/john-travoltage_en.html on an iPad2 (Tycho) and performance seems equivalent before and after.

**Comment by jessegreenberg** [comment]
@phet-steele this should be fixed, would you like to verify or are we good to close?

**Comment by jobara** [comment]
@jessegreenberg regarding your question about refreshing the virtual buffer, from what I understand is that it rebuilds the virtual buffer from the current DOM. I'm not sure how much of a problem this is anymore, but it used to be something that users would have to do if dynamic content wasn't showing up for them.

**Comment by jessegreenberg** [comment]
Thanks @jobara that is really good to know. For the hypothesis in https://github.com/phetsims/john-travoltage/issues/247#issuecomment-299995185, I would have expected refreshing to fix the problem.  This bug wasn't happening in the JAWS + Chrome combination, so perhaps it is an issue with IE11.

**Comment by jessegreenberg** [comment]
This is fixed, closing.
